server/test/repository/ApplicationRepositoryTest.java

```diff
@@ -63,56 +59,28 @@ public void submitApplication_updatesOtherApplicationVersions() {
         .isEqualTo(LifecycleStage.ACTIVE);
   }
 
-  @Test
-  public void submitApplication_doesNotUpdateOtherPrograms() {
-    Applicant one = saveApplicant("Alice");
-    Applicant two = saveApplicant("Bob");
-
-    Program pOne = saveProgram("Program");
-    Program pTwo = saveProgram("OtherProgram");
-
-    Application appOne = repo.createOrUpdateDraft(one, pOne).toCompletableFuture().join();
-
-    Application appTwo =
-        repo.submitApplication(two, pTwo, Optional.empty()).toCompletableFuture().join();
-
-    Instant appTwoInitialSubmitTime = appTwo.getSubmitTime();
-
-    assertThat(repo.getApplication(appOne.id).toCompletableFuture().join()).contains(appOne);
-    assertThat(repo.getApplication(appTwo.id).toCompletableFuture().join()).contains(appTwo);
-
-    // Submit application for different program.
-    repo.submitApplication(one, pOne, Optional.empty()).toCompletableFuture().join();
-
-    // Ensure the submit time didn't get changed when the application was set as obsolete.
-    assertThat(appTwo.getSubmitTime()).isEqualTo(appTwoInitialSubmitTime);
-  }
```

Comment:
I believe that this test was providing value, though the asserts could probably be improved. I believe the intention was that submitting an application for one program wouldn't have side-effects on applications for a separate program.

We could likely keep most of the test as-is, rename `appTwo` to `otherProgramApp`, then assert that `otherProgramApp` continues to have state `ACTIVE` and hasn't had the submit time modified.

Reply:
alright, it's back.